### ISSUE TITLE
make error: reference not defined in jack.go

### ISSUE THREAD

**Issue by sgmeiner** [andrepxx/go-dsp-guitar#10]
Running make fails:

```
meister@HP-II:~/.go/src/github.com/andrepxx/go-dsp-guitar$ make
GOPATH=`pwd`/../../../.. go build -o dsp -ldflags 'all=-w -s'
go: downloading github.com/andrepxx/go-jack v0.0.0-20220929171107-71a712d2f786
# github.com/andrepxx/go-dsp-guitar
/usr/local/go/pkg/tool/linux_amd64/link: running gcc failed: exit status 1
/tmp/go-link-156518109/000003.o: In Funktion `jack_set_port_rename_callback_go':
/home/meister/.go/pkg/mod/github.com/andrepxx/go-jack@v0.0.0-20220929171107-71a712d2f786/jack.go:49: Nicht definierter Verweis auf `jack_set_port_rename_callback'
collect2: error: ld returned 1 exit status

Makefile:29: die Regel für Ziel „dsp“ scheiterte
make: *** [dsp] Fehler 1

```
 The error text translates to
```
/home/meister/.go/pkg/mod/github.com/andrepxx/go-jack@v0.0.0-20220929171107-71a712d2f786/jack.go:49: Undefined reference to `jack_set_port_rename_callback'
...
Makefile:29: the rule for target "dsp" failed

```
Tried to install as proposed in https://github.com/andrepxx/go-dsp-guitar#building-the-software-from-source-locally

All commands except make ran successfully.

```
meister@HP-II:~/.go/src/github.com/andrepxx/go-dsp-guitar$ uname -a
Linux HP-II 4.15.0-142-generic #146~16.04.1-Ubuntu SMP Tue Apr 13 09:27:15 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

```

```
meister@HP-II:~/.go/src/github.com/andrepxx/go-dsp-guitar$ go env
GO111MODULE="auto"
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/meister/.cache/go-build"
GOENV="/home/meister/.config/go/env"
GOEXE=""
GOEXPERIMENT=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GOMODCACHE="/home/meister/.go/pkg/mod"
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/home/meister/.go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/local/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/linux_amd64"
GOVCS=""
GOVERSION="go1.20.2"
GCCGO="gccgo"
GOAMD64="v1"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/home/meister/.go/src/github.com/andrepxx/go-dsp-guitar/go.mod"
GOWORK=""
CGO_CFLAGS="-O2 -g"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-O2 -g"
CGO_FFLAGS="-O2 -g"
CGO_LDFLAGS="-O2 -g"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -Wl,--no-gc-sections -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build3210947670=/tmp/go-build -gno-record-gcc-switches"

```

```
meister@HP-II:~/.go/src/github.com/andrepxx/go-dsp-guitar$ echo $PATH
/home/meister:/home/meister/bin:/home/meister/.bin:/home/meister/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/usr/brlcad/bin:/usr/local/go/bin:/home/meister/.go/bin

```

What I've tried so far:
- installed libjack0 as it was required to install libjack-dev:

```
meister@HP-II:~/.go/src/github.com/andrepxx/go-dsp-guitar$ sudo apt-get install libjack-dev
...
The following packages have unfulfilled dependencies:
 libjack-dev : Depends on: libjack0 (= 1:0.124.1+20140122git5013bed0-3build2) but should not be installed.
```

- installed libjack-dev, as it was required for make

**Comment by andrepxx** [comment]
First, thank you for your interest in this software and your issue report.

I see your system is running on *Ubuntu 16.04 LTS*, which is almost 7 years old and "end of life" since April 30, 2021. I assume you have a good reason for that - for example, that newer releases do not run on your particular hardware. If this is not the case, I'd strongly suggest you update to a newer version of the distribution. "End-of-life" versions will no longer receive updates and at some point even the package repositories might be taken down so that you can no longer `apt-get` anything.

That aside: I cannot check this on my system, since I don't run *Ubuntu 16.04 LTS* anymore, but is there a way for you to update your `libjack-dev` package to version **0.125** at least? (It appears to be version **0.124** for you.) It's been around since 2016 and the function mentioned in the error message that appears during your build (`jack_set_port_rename_callback`) was added with version **0.125**.

This is the commit in the *JACK1* repo that added the callback: https://github.com/jackaudio/jack1/commit/0ac9ef7c813f4e67ab1e9a5fb4aea515fd9f1b31

Perhaps you could also update to `jackd2` and `libjack-jackd2-dev` instead of `jackd` and `libjack-dev`, in case these packages are available on *Ubuntu 16.04 LTS* and you don't have dependencies that force you to still use *JACK1* (instead of *JACK2*).

So, it's hard for me to check / "solve" this from here since I'm not working on your system, but the issue is basically that the version of JACK you have installed is too old. There are API changes from time to time and these currently cause the linking to fail.

Another "fix" one could try would be to vendor "andrepxx/go-jack", then remove support for `jack_set_port_rename_callback_go`. (It is implemented here: https://github.com/andrepxx/go-jack/blob/master/jack.go#L230) This should work, if this is the only "offending" API call, especially since this API method is not used by *go-dsp-guitar* at all. If the API changes are larger, this might not solve the problem, and it might be difficult to perform all the required changes.

Like I said, I cannot try this out, so any feedback on this issue is appreciated.

**Comment by sgmeiner** [comment]
Hi [andrepxx](https://github.com/andrepxx), thank you very much. I tried it on another computer with newer Ubuntu - there it works. I guess in time I will move everything to a newer Computer.
